runelite-client/src/main/java/net/runelite/client/plugins/raids/bats/BatsLocator.java

```diff
@@ -0,0 +1,503 @@
+					addSolutionSet(chests, 14, 19, 32, 73);
+					addSolutionSet(chests, 15, 21, 39, 48);
+					addSolutionSet(chests, 17, 19, 44, 47);
+					addSolutionSet(chests, 24, 53, 61, 63);
+					addSolutionSet(chests, 26, 53, 69, 71);
+					addSolutionSet(chests, 26, 53, 69);
+					addSolutionSet(chests, 27, 29, 57, 67);
+					addSolutionSet(chests, 27, 36, 70);
+					addSolutionSet(chests, 28, 31, 43, 71);
+					addSolutionSet(chests, 30, 43, 64, 72);
+					addSolutionSet(chests, 30, 54);
+					addSolutionSet(chests, 32, 34, 61, 66);
+					addSolutionSet(chests, 33, 42, 55, 65);
+					addSolutionSet(chests, 33, 46, 68, 73);
+					addSolutionSet(chests, 35, 37, 56, 63);
+					addSolutionSet(chests, 35, 38, 48, 58);
+					addSolutionSet(chests, 39, 51, 59, 65);
+					addSolutionSet(chests, 40, 47, 60, 62);
+					addSolutionSet(chests, 41, 50, 54, 70);
+					addSolutionSet(chests, 42, 45, 52, 58);
+					addSolutionSet(chests, 44, 50, 54, 66);
+					break;
+				case STRAIGHT:
+					addSolutionSet(chests, 0, 39, 43, 51);
+					addSolutionSet(chests, 1, 15, 20, 53);
+					addSolutionSet(chests, 2, 10, 42, 44);
+					addSolutionSet(chests, 3, 21, 54);
+					addSolutionSet(chests, 4, 14, 38, 52);
+					addSolutionSet(chests, 5, 6, 35, 41);
+					addSolutionSet(chests, 7, 16, 34, 49);
+					addSolutionSet(chests, 9, 12, 26, 27);
+					addSolutionSet(chests, 13, 25, 30, 31);
+					addSolutionSet(chests, 15, 20, 53);
+					addSolutionSet(chests, 17, 24, 34, 58);
+					addSolutionSet(chests, 18, 23, 35, 57);
+					addSolutionSet(chests, 19, 26, 47, 65);
+					addSolutionSet(chests, 21, 33, 36, 61);
+					addSolutionSet(chests, 22, 25, 46, 55);
+					addSolutionSet(chests, 24, 34, 58);
+					addSolutionSet(chests, 28, 40, 52, 63);
+					addSolutionSet(chests, 29, 41, 42, 64);
+					addSolutionSet(chests, 30, 32, 37, 62);
+					addSolutionSet(chests, 39, 43, 51);
+					addSolutionSet(chests, 43, 45, 50, 60);
+					addSolutionSet(chests, 51, 53, 56, 59);
+					break;
+			}
+
+			for (Chest chest : this.chests.values())
+			{
+				if (solutionSetsContaining(chest).size() == 0)
+				{
+					openChest(chest, Chest.State.GRUBS);
+				}
+			}
+			for (Chest chest : poisonBatsChests)
+			{
+				openChest(chest, Chest.State.POISON);
+			}
+			for (Chest chest : grubsChests)
+			{
+				openChest(chest, Chest.State.GRUBS);
+			}
+
+			findSolutionSetCounts();
+			drawChestStates = true;
+		}
+	}
+
+	private void addSolutionSet(ArrayList<Chest> chests, int... indices)
+	{
+		ArrayList<Chest> solutionSet = new ArrayList<>();
+		for (int index : indices)
+		{
+			solutionSet.add(chests.get(index));
+		}
+		solutionSets.add(solutionSet);
+	}
+
+	public void poisonSplatEvent(WorldPoint worldPoint)
+	{
+		Chest chest = chests.get(worldPoint);
+		if (chest != null)
+		{
+			chest.setTickPoison(client.getTickCount());
+		}
+	}
+
+	//This does have a delay switching from poison chest to bats chest which can be observed by the color change at the chest, only when starting the plugin.
```

Comment:
It looks like you have the same comment here twice, though I don't really understand either of them.

Reply:
A color change from green (poison) to white (bats) is observerd when the raids plugin is started. The bats chest can't be differentiated from the poison chests so it is coloured green when it spawns. The next tick arrives and no poison splat has occured so the bats locator thinks it is a bats chest. It could happen that the bats chest and two poison chests are open when the raids plugin is started which results in white dots, even for the poison chests. It is ofcourse unlikely that the raids plugin is switched on during a raid but I thought it was worth mentioning.